### ISSUE TITLE
Changes in error object during error is active

### ISSUE THREAD

**Issue by philipkirmse** [VDA5050/VDA5050#337]
Based on VDA5050 it is defined that an error/information object will stay in error/info array in state messages (in each message) until it is resolved on AGV side.

On MC side this information is usually used to create a notification event and will be stored in archive. 

In our opinion an error object, information object need to be unchanged in each state per event until it is resolved on AGV side for all fields per error object/message including all reference field. If the error object is changed e.g with dynamic fields in errorReferences etc. there are several challenges  with general implementation, for example

- interpretation of a new error object, when content is changed
- archive only one error object per error event.

In our opinion there are following gaps in definition:

- an error/info object should not be changed in content in AGV state until it is resolved on AGV side
- to fill up the need to send an visualization data with dynamic values (like a error timer) there should be a sep. field for only this Data for visualization, but not for archiving with the rest of the notification messages.

Can we discuss this enhancements? Or how should error object be interpreted. Is it valid to change error reference values until the error is not finally resolved?

**Comment by camil22** [comment]
I dont think that MC can differentiate between a new error with same errorType and different references and the same error with changed references without introducing some identifier or error timestamp. That said the state message reflects the current state of an agv not a specific event.

**Comment by philipkirmse** [comment]
Yes, Not as easy as it would be with an identifier but it is possible, when it is not changed the object dynamically. 

Use cases are:

- to define start and endtime of an event, and so duration
-Distinguish between different message objects by event
- store it only once
- Filtered visualization cases etc.
